src/components/NewItemForm.jsx

```diff
@@ -55,14 +103,111 @@ const NewItemForm = () => {
   const [endpoint, setEndpoint] = useState("");
   const [password, setPassword] = useState("");
   const [username, setUserName] = useState("");
+  const [itemNameError, setItemNameError] = useState(clearError);
+  const [endpointError, setEndpointError] = useState(clearError);
+  const [passwordError, setPasswordError] = useState(clearError);
+  const [usernameError, setUserNameError] = useState(clearError);
+
   const [collectionID, setCollectionID] = useState("");
+  const [collectionArray, setCollectionArray] = useState([]);
+  const [defaultCollectionsID, setDefaultCollectionID] = useState("");
+
+  const [symmetricKey, setSymmetricKey] = useState("");
 
   const handleCollectionChange = (event) => {
     setCollectionID(event.target.value);
   };
 
-  const onSaveVaultItem = () => {
-    console.log("status");
+  const getUserItems = async () => {
+    try {
+      let collectionResponse = await getCollections(getCollections);
+      let keyResponse = await getUserKey();
+      setCollectionArray(collectionResponse.data);
+
+      // We need to decrypt this in the future
+      let encryptedSymmetricKey = keyResponse.data.encrypted_symmetric_key;
+      for (let i = 0; i < collectionResponse.data.length; i++) {
+        let collection = collectionResponse.data[i];
+        if (collection.name == "Default") {
+          setDefaultCollectionID(collection.uuid);
+          setCollectionID(collection.uuid);
+        }
+      }
+    } catch (error) {
+      networkErrorHandler(error);
+    }
+  };
+
+  useEffect(() => {
+    getUserItems();
+  }, []);
+
+  const onSaveVaultItem = async () => {
+    let errorStatus = false;
+    errorStatus = isNullCheck(itemName, setItemNameError);
+    errorStatus = isNullCheck(password, setPasswordError);
+    errorStatus = isNullCheck(endpoint, setEndpointError);
+    errorStatus = isNullCheck(username, setUserNameError);
+
+    if (errorStatus) {
+      return;
+    }
+
+    let vaultItemObject = {
+      name: itemName,
+      url: endpoint,
+      username: username,
+      password: password,
+    };
+
+    console.log(vaultItemObject);
+    // try {
+    //   let response = await postVaultItem(
+    //     vaultItemObject,
+    //     symmetricKey,
+    //     collectionID
+    //   );
+    // } catch (error) {
+    //   networkErrorHandler(error);
+    // }
+  };
+
+  const isNullCheck = (field, errorHandler) => {
+    if (field === "") {
+      errorHandler(requiredFieldsError);
+      return true;
+    }
+    return false;
+  };
+
+  const clearAllErrors = () => {
+    setCollectionError(clearError);
+    setItemNameError(clearError);
+    setEndpointError(clearError);
+    setPasswordError(clearError);
+    setUserNameError(clearError);
+  };
+
+  const clearAllInputs = () => {
+    setItemName("");
+    setEndpoint("");
+    setPassword("");
+    setUserName("");
+    setCollectionName("");
```

Comment:
We also need an additional call to set the default collection id. In the vault items form it is still showing the last selected folder when I open the modal again.